### PR TITLE
Add an additional option in the configuration of module

### PR DESCRIPTION
**Pull request by Xiony**
Add an additional option in the configuration of module :
 "Allow a comment, only if the customer has already ordered the product"

If the "Allow guest reviews" option is activated, this automatically deactivates the option so as not to be in conflict.

It is the first time that I participate on a project, I have doubts about a modification that I made for the upgrade in : upgrade/install-5.0.1.php I hope to have done well. 
I don't know where the version numbers come from.

Another question, I wanted to add a translation for French, but apparently it is present directly in prestashop and not in the module.

### CODE REVIEW

File: productcomments.php [hunk 1]
```diff
@@ -579,6 +603,7 @@ public function getConfigFieldsValues()
             'PRODUCT_COMMENTS_USEFULNESS' => Tools::getValue('PRODUCT_COMMENTS_USEFULNESS', Configuration::get('PRODUCT_COMMENTS_USEFULNESS')),
             'PRODUCT_COMMENTS_MINIMAL_TIME' => Tools::getValue('PRODUCT_COMMENTS_MINIMAL_TIME', Configuration::get('PRODUCT_COMMENTS_MINIMAL_TIME')),
             'PRODUCT_COMMENTS_COMMENTS_PER_PAGE' => Tools::getValue('PRODUCT_COMMENTS_COMMENTS_PER_PAGE', Configuration::get('PRODUCT_COMMENTS_COMMENTS_PER_PAGE')),
+            'PRODUCT_COMMENTS_ALLOW_ALREADY_ORDERED' => Tools::getValue('PRODUCT_COMMENTS_ALLOW_ALREADY_ORDERED', Configuration::get('PRODUCT_COMMENTS_ALLOW_ALREADY_ORDERED')),
```
Comment (PierreRambaud) [inline productcomments.php:606]:
```suggestion
            'PRODUCT_COMMENTS_ALLOW_ALREADY_ORDERED' => (bool) Tools::getValue('PRODUCT_COMMENTS_ALLOW_ALREADY_ORDERED', Configuration::get('PRODUCT_COMMENTS_ALLOW_ALREADY_ORDERED')),
```

File: src/Repository/ProductCommentRepository.php [hunk 2]
```diff
@@ -336,4 +344,31 @@ private function getLastComment(array $criteria)
...
+                ->where('o.id_customer = :id_customer')
+                ->andWhere('od.product_id = :id_product')
+                ->setParameter('id_customer', (int) $idCustomer)
+                ->setParameter('id_product', (int) $idProduct)
+            ;
+            $result = $qb->execute()->fetchAll();
+            return count($result)>0;
+        }
+        return true;
+    }
```
Comment (PierreRambaud) [inline src/Repository/ProductCommentRepository.php:373]:
```suggestion
        if (!$this->customerOrdererProductAllowed) {
            return true;
        }
        
        //Test if the customer has a product ordered
        /** @var QueryBuilder $qb */
        $qb = $this->connection->createQueryBuilder();
        $qb
            ->addSelect('od.product_id')
            ->from($this->databasePrefix . 'order_detail', 'od')
            ->leftJoin('od', $this->databasePrefix . 'orders', 'o', 'od.id_order = o.id_order')
            ->where('o.id_customer = :id_customer')
            ->andWhere('od.product_id = :id_product')
            ->setParameter('id_customer', (int) $idCustomer)
            ->setParameter('id_product', (int) $idProduct);

        return $qb->execute()->rowCount() > 0;
    }
```

**Comment by Xiony** [comment]
Thanks for the review.

however for the translation,
on file productComments.php line 332:

`$this-> trans ('Allow a comment, only if the customer has already ordered the product', [], 'Modules.Productcomments.Admin')`

Where can I define it in French?

**Comment by sarahdib** [comment]
Hello @Xiony 

Could you target the dev branch ?

Thank you!

**Comment by PierreRambaud** [comment]
Could you please rebase your branch?

**Comment by matks** [comment]
Hi,

Since we had no news for almost 1 year, I will follow [our guidelines](https://devdocs.prestashop.com/1.7/project/maintainers-guide/reviewing-pull-requests/#stale-pull-requests) and I'll close this pull request as it seems stale. Feel free to reopen or open another one if you think it's still relevant.

Thanks!

**Pull request closed without merging**